internal/alertmanager/models.go

```diff
@@ -92,7 +101,7 @@ func (am *Alertmanager) pullSilences(version string) error {
 	}
 
 	start := time.Now()
-	silences, err := mapper.GetSilences(am.URI, am.RequestTimeout)
+	silences, err := mapper.GetSilences(am.URI, am.RequestTimeout, am.TLSConfig)
```

Comment:
Why are you having to thread TLSConfig? I'd expect the transport to be allocated (with the TLS config) when the AM was added.

Reply:
yeah, that's not great, refactored to pass around tls.Config instead
anything else stands out here?